server/container/guice/jmx/src/main/java/org/apache/james/modules/server/JmxConfiguration.java

```diff
@@ -20,35 +20,48 @@
 package org.apache.james.modules.server;
 
 import java.util.Objects;
+import java.util.Optional;
 
 import org.apache.commons.configuration.PropertiesConfiguration;
 import org.apache.james.util.Host;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
 
 public class JmxConfiguration {
 
     public static final String LOCALHOST = "localhost";
     public static final int DEFAULT_PORT = 9999;
 
-    public static final JmxConfiguration DEFAULT_CONFIGURATION = new JmxConfiguration(Host.from(LOCALHOST, DEFAULT_PORT));
+    public static final JmxConfiguration DEFAULT_CONFIGURATION = new JmxConfiguration(Optional.of(Host.from(LOCALHOST, DEFAULT_PORT)));
+    public static final JmxConfiguration DISABLED = new JmxConfiguration(Optional.empty());
 
     public static JmxConfiguration fromProperties(PropertiesConfiguration configuration) {
+        boolean jmxEnabled = configuration.getBoolean("jmx.enabled", true);
+        if (!jmxEnabled) {
+            return DISABLED;
+        }
+
         String address = configuration.getString("jmx.address", LOCALHOST);
         int port = configuration.getInt("jmx.port", DEFAULT_PORT);
-        return new JmxConfiguration(Host.from(address, port));
+        return new JmxConfiguration(Optional.of(Host.from(address, port)));
     }
 
-    private final Host host;
+    private final Optional<Host> host;
 
     @VisibleForTesting
-    JmxConfiguration(Host host) {
+    JmxConfiguration(Optional<Host> host) {
         this.host = host;
     }
 
+    public boolean isEnabled() {
+        return host.isPresent();
```

Comment:
this is strange. I'd prefer to really have a boolean in the configuration. Optional for host could have meant configured / not configured (even if for now a default configuration is loaded).